File: dkr/1/lab_1.pas
uses math;

procedure task_two(start_: real; end_: integer; step: real; func: real -> real);
begin
  writeln('|   x   |   y   |');
  writeln('¦---------------¦');
  
  var x := start_;
  while x <= end_ do
  begin
    var y: real;
    if x > 0 then
      y := func(x)
    else
      y := func(abs(x));
    writeln('¦', x:6:2, ' |', y:7:2, '¦');
    x += step;
  end;
  
  writeln('¦---------------¦');
end;


function task_one(x: integer): real;
begin
  var func: real -> real;
  var doTaskTwo: string := 'n';
  
  if x < -9 then
  begin
    func := x -> ln(x) / (x*x);
    writeln(func(x));
  end
  else if (x >= -9) and (x < -2) then
  begin
    func := x -> power(x, 1/3) * x * (-1);
    writeln(func(x));
  end
  else if (x >= -2) and (x < 4) then
  begin
    func := x -> ln(x) + power(x, 0.1*x);
    writeln(func(x));
  end
  else if (4 <= x) then
  begin
    func := x -> ln(x) / sqr(x);
    writeln(func(x));
  end
  else
    func := x -> 0;
  
  writeln('Выполнить второе задание? (y/n, default - n)');
  readln(doTaskTwo);
  if doTaskTwo = 'y' then
    task_two(-11, 6, 0.2, func);
end;


begin
  writeln(task_one(5));
end.

File: dkr/1/lab_1.py
from math import log, cbrt
from typing import Callable


def two(start: float, end: float, step: float,
        f: Callable[[float], float]) -> None:
    x = start

    print("-" * 20)
    print(f'|{"x":^8}|{"y":^9}|')
    print("-" * 20)

    while x <= end:
        try:
            y = f(x)
            print(f'| {x:6.2f} | {y:7.2f} |')
        except (ValueError, ZeroDivisionError):
            print(f'| {x:6.2f} | {f(abs(x)):>7} |')
        x += step
    print("-" * 20)


def one(x: float) -> Callable[[float], float]:
    if x < -9:
        return lambda x: log(x) / (x**2)
    elif -9 < x < -2:
        return lambda x: -x * cbrt(x)
    elif -2 <= x < 4:
        return lambda x: log(x) + x**(x*0.1)
    elif x >= 4:
        return lambda x: log(x) / x**2
    else:
        return lambda x: 0


def main():
    x = float(input("Введите x: "))
    func = one(x)
    print(f"Ответ на первое задание: {func(x)}")

    if input('Выполнить второе задание? (y/n, default - n)\n').lower() == 'y':
        two(start=-11.0, end=6.0, step=0.2, f=func)


main()